src/bin/miri.rs

```diff
@@ -106,14 +106,16 @@ impl rustc_driver::Callbacks for MiriBeRustCompilerCalls {
             // Queries overriden here affect the data stored in `rmeta` files of dependencies,
             // which will be used later in non-`MIRI_BE_RUSTC` mode.
             config.override_queries = Some(|_, local_providers, _| {
-                // `exported_symbols()` provided by rustc always returns empty result if
-                // `tcx.sess.opts.output_types.should_codegen()` is false.
+                // `exported_symbols` and `reachable_non_generics` provided by rustc always returns
+                // an empty result if `tcx.sess.opts.output_types.should_codegen()` is false.
                 local_providers.exported_symbols = |tcx, cnum| {
                     assert_eq!(cnum, LOCAL_CRATE);
                     tcx.arena.alloc_from_iter(
                         // This is based on:
                         // https://github.com/rust-lang/rust/blob/2962e7c0089d5c136f4e9600b7abccfbbde4973d/compiler/rustc_codegen_ssa/src/back/symbol_export.rs#L62-L63
                         // https://github.com/rust-lang/rust/blob/2962e7c0089d5c136f4e9600b7abccfbbde4973d/compiler/rustc_codegen_ssa/src/back/symbol_export.rs#L174
+                        // FIXME not sure why we need this `allow`, the same code in rustc seems to be accepted?
+                        #[allow(rustc::potential_query_instability)]
                         tcx.reachable_set(()).iter().filter_map(|&local_def_id| {
```

Comment:
@davidtwco this one is surprising to me. This is the same code as [in rustc](https://github.com/rust-lang/rust/blob/6098e795a20f03654ab967a3705a495454abb05b/compiler/rustc_codegen_ssa/src/back/symbol_export.rs#L60-L62), so why does the lint fire here but not there?

Reply:
Oh, it is being allowed [here](https://github.com/rust-lang/rust/blob/6098e795a20f03654ab967a3705a495454abb05b/compiler/rustc_codegen_ssa/src/lib.rs#L11).